resource/resource.go

```diff
@@ -118,6 +119,22 @@ func FromYamlByte(data []byte) ([]Resource, error) {
 	return results, nil
 }
 
+// expandEnv replaces ${var} or $var in config according to the values of the current environment variables.
+// The replacement is case-sensitive. References to undefined variables are replaced by the empty string.
+// A default value can be given by using the form ${var:-default value}.
+func expandEnv(data []byte) []byte {
+	return []byte(os.Expand(string(data), func(key string) string {
+		keyAndDefault := strings.SplitN(key, ":-", 2)
+		key = keyAndDefault[0]
+
+		v := os.Getenv(key)
+		if v == "" && len(keyAndDefault) == 2 {
```

Comment:
🤔  I think this would be better if it failed when the env var is missing?
Behaviour with *nix tools is patchy in this regard, but unless you define a default I think its better to fail if the env var is missing or blank (as I would assume if you set one without a default, you mean to use it!)

Reply:
I think I can use https://pkg.go.dev/os#LookupEnv for that

Reply:
but it's seems that because the mapping function doesn't handle any possibility of error I think we will have to stick with the current implementation

Reply:
yes I also agree that this should crash when an env var is missing or empty, except if a default value is provided. 
so in case of missing or empty env, if default value, use it, if not crash

Reply:
in most of the tool they are defaulting to an empty string so I don't know if we should raise an error

Reply:
I personally have many issues where an env was not set properly and was used in some config like this and end up pushing incomplete payload to APIs breaking things or worse silently doing nothing leading to time wasted to debug it.

Reply:
Conduktor Console YAML configuration also supports env var expansion and crash at startup if referenced env var in YAML are not defined or empty.

Reply:
to do it right it will need a more work because of go lib limitation

Reply:
LookupEnv should do the trick for us no ? like 
```go
val, ok := os.LookupEnv(key)
if ok {
  return v
} else {
  if  len(keyAndDefault) == 2 {
    return  keyAndDefault[1]
  } else {
    panic(fmt.Sprintf("Missing Environment variable %v", key))
  }
}
```

Reply:
yes but you will report only one error on Missing Environment variable, and you will crash for all the YAML files you try to apply

Reply:
Given its a one shot tool anyway (its not a long running process with needs to stay up), I would agree we can just use `panic` to exit in this case. It is kind of a panic scenario!

Reply:
> yes but you will report only one error on Missing Environment variable, and you will crash for all the YAML files you try to apply

ultimately we would need some sort of pre-step to validate all the files first and weed out the total set of problems (env vars or otherwise). If we don't have that today - then while I would agree with @bthuillier  that we _should_ have nice error handling, I also agree with @qboileau  that if we are going to have env vars, it has to fail if they are missing. So if we add them now, its the `panic` route - or else a bigger refactor to pre-check all the configs for problems (which feels like a load more work?)

FWIW I am happy from my side with adding the env vars, and using panic as soon as a missing one if found.